Production_scripts/Process_extracts/extract_delayed_discharges.R

```diff
@@ -0,0 +1,95 @@
+#####################################################
+# Draft pre processing code for Delayed Discharges
+# Author: Jennifer Thom
+# Date: April 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input - Jul16_{latest dd period}DD_LinkageFile.zssav
+# Description - Preprocessing of raw delayed discharges file.
+#               Tidy up file in line with SLF format
+#                prior to processing.
+#####################################################
+
+# Load Packages #
+library(dplyr)
+library(janitor)
+library(lubridate)
+library(createslf)
+
+
+# Read in data---------------------------------------
+
+year <- check_year_format("1920")
+
+dd_file <- haven::read_sav(get_dd_path(ext = "zsav")) %>%
+  clean_names() %>%
+  # rename variables
+  rename(
+    keydate1_dateformat = rdd,
+    keydate2_dateformat = delay_end_date
+  )
+
+
+# Data Cleaning---------------------------------------
+
+dd_clean <- dd_file %>%
+  # Use end of the month date for records with no end date (but we think have ended)
+  # Create a flag for these records
+  mutate(
+    month_end = lubridate::ceiling_date(keydate1_dateformat, "month") - 1,
+    ammended_dates = case_when(
+      keydate2_dateformat == ymd("1900,1,1") ~ TRUE,
+      TRUE ~ FALSE
+    ),
+    keydate2_dateformat = if_else(ammended_dates == TRUE, month_end, keydate2_dateformat)
+  ) %>%
+  # Drop any records with obviously bad dates
+  filter(
+    (keydate1_dateformat <= keydate2_dateformat) | is.na(keydate2_dateformat)
+  ) %>%
+  # set up variables
+  mutate(
+    recid = "DD",
+    smrtype = "DelayedDis",
+    year = year
+  ) %>%
+  # recode blanks to NA
+  mutate(
+    primary_delay_reason = na_if(primary_delay_reason, ""),
+    secondary_delay_reason = na_if(secondary_delay_reason, "")
+  ) %>%
+  # create flags for no_end_date and correct_dates
+  create_dd_flags(year) %>%
+  # Keep only records which have an end date (except Mental Health) and fall within our dates.
+  filter(correct_dates == TRUE & no_end_date == FALSE)
+
+
+## save outfile ---------------------------------------
+outfile <- dd_clean %>%
+  select(
+    year,
+    recid,
+    original_admission_date,
+    keydate1_dateformat,
+    keydate2_dateformat,
+    SMRType,
```

Comment:
This column is actually smrtype, so maybe 
```suggestion
    SMRType = smrtype,
```

Reply:
We were a bit hit and miss in SPSS with the variable name case (because SPSS doesn't care) but for R, we should keep all variables lower case - so this will have to be `smrtype`